rest/data/distro.go

```diff
@@ -74,3 +78,49 @@ func DeleteDistroById(distroId string) error {
 	}
 	return nil
 }
+
+type CopyDistroOpts struct {
+	DistroIdToCopy string
+	NewDistroId    string
+}
+
+func CopyDistro(ctx context.Context, env evergreen.Environment, u *user.DBUser, opts CopyDistroOpts) error {
+	distroToCopy, err := distro.FindOneId(opts.DistroIdToCopy)
+	if err != nil {
+		return errors.Wrapf(err, "finding distro '%s'", opts.DistroIdToCopy)
+	}
+	if distroToCopy == nil {
+		return gimlet.ErrorResponse{
+			StatusCode: http.StatusNotFound,
+			Message:    fmt.Sprintf("distro '%s' not found", opts.DistroIdToCopy),
+		}
+	}
+
+	distroToCopy.Id = opts.NewDistroId
+	return newDistro(ctx, env, distroToCopy, u)
+}
+
+func newDistro(ctx context.Context, env evergreen.Environment, d *distro.Distro, u *user.DBUser) error {
+	settings, err := evergreen.GetConfig()
```

Comment:
I'm a bit curious - is there anything that we need from the DB's config here for validation? It seems like `env` was possibly passed in to use `env.Settings()`.

Either way, `env` is currently an unused parameter.

Reply:
The validator uses the provided `env.Settings()` to check that the distro has valid SSH keys. But good catch, I removed it as a param here.